src/tensors/gpu/prod.cpp

```diff
@@ -483,6 +430,45 @@ void CSRProd(marian::Tensor C,
     St_indices = allocator->alloc<int>(numValues);
     St_offsets = allocator->alloc<int>(colsS + 1);
     // transpose the second argument
+#if CUDA_VERSION >= 11000
+    size_t buffer_size;
+    CUSPARSE_CHECK(cusparseCsr2cscEx2_bufferSize(cusparseHandle,
+                                          /*m=*/ rowsS, // number of rows of matrix
+                                          /*n=*/ colsS, // number of columns of matrix
+                                          /*nnz=*/ (int)numValues,
+                                          /*csrcVal=*/          S_values ->data<float>(),
+                                          /*csrcRowPtr=*/ (int*)S_offsets->data<IndexType>(),
+                                          /*csrcColInd=*/ (int*)S_indices->data<IndexType>(),
+                                          /*cscVal=*/    St_values ->data<float>(),  // transposed version goes here
+                                          /*cscColPtr=*/ St_offsets->data<int>(),
+                                          /*cscRowInd=*/ St_indices->data<int>(),
+                                          /*valType*/ CUDA_R_32F,
+                                          /*copyValues=*/ CUSPARSE_ACTION_NUMERIC,
+                                          /*idxBase=*/ CUSPARSE_INDEX_BASE_ZERO,
+                                          /*alg*/ CUSPARSE_CSR2CSC_ALG1,
+                                          /*bufferSize*/ &buffer_size));
+    MemoryPiece::PtrType buffer= (buffer_size > 0) ? allocator->alloc<uint8_t>(buffer_size) : nullptr;
+
+    CUSPARSE_CHECK(cusparseCsr2cscEx2(cusparseHandle,
+                                          /*m=*/ rowsS, // number of rows of matrix
+                                          /*n=*/ colsS, // number of columns of matrix
+                                          /*nnz=*/ (int)numValues,
+                                          /*csrcVal=*/          S_values ->data<float>(),
+                                          /*csrcRowPtr=*/ (int*)S_offsets->data<IndexType>(),
+                                          /*csrcColInd=*/ (int*)S_indices->data<IndexType>(),
+                                          /*cscVal=*/    St_values ->data<float>(),  // transposed version goes here
+                                          /*cscColPtr=*/ St_offsets->data<int>(),
+                                          /*cscRowInd=*/ St_indices->data<int>(),
+                                          /*valType=*/ CUDA_R_32F,
+                                          /*copyValues=*/ CUSPARSE_ACTION_NUMERIC,
+                                          /*idxBase=*/ CUSPARSE_INDEX_BASE_ZERO,
+                                          /*alg=*/ CUSPARSE_CSR2CSC_ALG1,
+                                          /*buffer=*/ buffer->data<uint8_t>()));
+
+    if (buffer)
+      allocator->free(buffer);
+    ABORT("This code is untested. Please remove this ABORT once tests exist and pass.");
```

Comment:
How about the tests mentioned in the comments to the PR in `operator_tests.cpp`?

Reply:
Ah, saw the comment concerning the tests.